pkg/csi/cinder/controllerserver.go

```diff
@@ -318,7 +318,35 @@ func (cs *controllerServer) GetCapacity(ctx context.Context, req *csi.GetCapacit
 }
 
 func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("ControllerExpandVolume is not yet implemented"))
+	klog.V(4).Infof("ControllerExpandVolume: called with args %+v", *req)
+
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
+	}
```

Comment:
should we add volume existence check here also?

Reply:
https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerexpandvolume-errors

Reply:
yes, will add . Thanks